solid/src/main/java/com/inrupt/client/solid/SolidClient.java

```diff
@@ -280,14 +285,15 @@ public <T extends Resource> CompletionStage<Void> delete(final T resource, final
         defaultHeaders.firstValue(USER_AGENT).ifPresent(agent -> builder.setHeader(USER_AGENT, agent));
         headers.firstValue(USER_AGENT).ifPresent(agent -> builder.setHeader(USER_AGENT, agent));
 
-        return client.send(builder.build(), Response.BodyHandlers.ofByteArray()).thenApply(res -> {
-            if (isSuccess(res.statusCode())) {
-                return null;
-            } else {
-                throw SolidClientException.handle("Unable to delete resource", resource.getIdentifier(),
-                        res.statusCode(), res.headers(), new String(res.body(), UTF_8));
-            }
-        });
+        return client.send(
+            builder.build(),
+            Response.BodyHandlers.throwOnError(
+                Response.BodyHandlers.ofByteArray(),
+                (r) -> Response.isSuccess(r.statusCode()),
+                httpExceptionMapper("Unable to delete resource.")
+            )
+        )
+        .thenAccept(o -> { /* no-op */ });
```

Comment:
What does this thenAccept no-op do? Is it needed?

Reply:
It is needed to honor the function signature. I had a similar doubt/question :) : https://github.com/inrupt/solid-client-java/pull/1157#discussion_r1568766232